server/src/main/java/com/dut/healthme/service/impl/RestaurantServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package com.dut.healthme.service.impl;
+
+import com.dut.healthme.common.constant.ErrorMessageConstants;
+import com.dut.healthme.common.exception.BadRequestException;
+import com.dut.healthme.entity.Restaurant;
+import com.dut.healthme.entity.enums.RestaurantStatus;
+import com.dut.healthme.repository.RestaurantsRepository;
+import com.dut.healthme.service.RestaurantService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+@Service
+@RequiredArgsConstructor
+public class RestaurantServiceImpl implements RestaurantService {
+    private final RestaurantsRepository restaurantsRepository;
+    @Override
+    public List<Restaurant> getRestaurantsWaiting() {
+        return this.restaurantsRepository.findByStatus(RestaurantStatus.AWAITING_APPROVAL);
```

Comment:
Ở service này lên lấy ra cả những nhà hàng có status là APPROVAL_FAILED luôn nha. Lỡ admin có từ chối nhầm thì họ vẫn có thể chấp nhận lại được.
`Note: Chỉnh lại repository rồi mới chỉnh ở service nha`
```suggestion
        return this.restaurantsRepository.findByStatusIn(List.of(RestaurantStatus.AWAITING_APPROVAL, RestaurantStatus.APPROVAL_FAILED));
```